networkx/algorithms/broadcasting.py

```diff
@@ -66,12 +68,10 @@ def tree_broadcast_center(G):
     """
     # Assert that the graph G is a tree
     if not nx.is_tree(G):
-        NetworkXError("Input graph is not a tree")
+        raise nx.NetworkXError("input graph is not a tree")
     # step 0
-    if G.number_of_nodes() == 2:
-        return 1, set(G.nodes())
-    if G.number_of_nodes() == 1:
-        return 0, set(G.nodes())
+    if len(G) in {1, 2}:
+        return len(G) - 1, set(G.nodes())
```

Comment:
Since this comes after the `is_tree` check, you could in principle do away with the 1, 2 checking entirely

```suggestion
    if (n := len(G)) < 3:
        return n - 1, set(G)
```

Not even better IMO, just different :upside_down_face:

Reply:
I like it!